_posts/2020-05-07-health-groups.adoc

```diff
@@ -0,0 +1,235 @@
+= Health Groups - experimental MicroProfile Health extension in SmallRye
+
+If you are reading this blog, you have probably already heard about
+https://github.com/eclipse/microprofile-health[Eclipse MicroProfile Health]
+technology. In this article, we will introduce a new experimental extension to
+MicroProfile Health 2.2 called Health groups.
+
+All of the source code that we will present in this article is available at this
+GitHub repository https://github.com/xstefank/health-groups.
+
+== MicroProfile 2.2 health checks
+
+As you know, to define a health check procedure in MicroProfile Health you need
+something like this:
+
+[source,java]
+----
+package io.xstefank;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.Readiness;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped // 1. CDI scope
+@Readiness         // 2. Qualifier (Readiness, Liveness, Health)
+public class ReadinessHealthCheck implements HealthCheck {   // 3. implements HealthCheck interface
+    
+    @Override
+    public HealthCheckResponse call() {
+        return HealthCheckResponse.up(ReadinessHealthCheck.class.getSimpleName());
+    }
+}
+----
+
+So just to recap, to define a health check your class:
+
+- needs to be a CDI bean
+- needs to have one of the health qualifiers (Readiness, Liveness, or deprecated Health)
+- must implement the HealthCheck interface
+
+This is everything that you can use in MicroProfile 2.2. But what if you have a
+use case that doesn't fall into one of Readiness or Liveness categories? The answer
+is custom health groups.
+
+== Health Groups
+
+SmallRye Health 2.2.0 (already included in Quarkus 1.4.0) comes with a new addition to this API 
+called health groups. A health group is basically a custom group of health check 
+procedures that the user defines based on individual needs. As you probably noticed,
+Readiness and Liveness health checks already represent two predefined health groups. 
+With the health groups feature you can define as many additional health check groups
+as you need.
+
+=== Defining a health Group
+
+To define a health group you need only to declare at least one health check
+procedure that belongs to the health group:
+
+[source,java]
+----
+package io.xstefank;
+
+import io.smallrye.health.HealthGroup;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+@HealthGroup("custom-group-1")
+public class CustomGroup1Check implements HealthCheck {
+    
+    @Override
+    public HealthCheckResponse call() {
+        return HealthCheckResponse.up(CustomGroup1HC.class.getSimpleName());
+    }
+}
+----
+
+So the only difference now is the change of the qualifier for which we now use
+`io.smallrye.health.HealthGroup`. This qualifier takes a string argument
```

Comment:
Just thinking -- do you perhaps need an extra dependency to get this class? (I'd expect it to come from something like `io.smallrye:smallrye-health-api`.)

Reply:
yes, we have https://search.maven.org/artifact/io.smallrye/smallrye-health-extension-api/2.2.1/jar. But this is included in Quarkus extension.

Reply:
Should this be added to the blog post? I use only reference to SR Health 2.2.0.

Reply:
I don't know, your call :-)

Reply:
Added, I think it's a good idea. Thanks.